src/imgui/ImGuiReverseBar.cc

```diff
@@ -62,23 +62,26 @@ void ImGuiReverseBar::showMenu(MSXMotherBoard* motherBoard)
 		}
 		ImGui::Separator();
 
-		im::Menu("Load state ...", [&]{
+		loadStateOpen = im::Menu("Load state ...", [&]{
+			if (!loadStateOpen) {
+				// on each re-open of this menu, we recreate the list of states
+				stateNames.clear();
+				for (auto context = userDataFileContext(STATE_DIR);
+				     const auto& path : context.getPaths()) {
+					foreach_file(path, [&](const std::string&, std::string_view name) {
+						if (name.ends_with(STATE_EXTENSION)) {
+							name.remove_suffix(STATE_EXTENSION.size());
+							stateNames.emplace_back(std::string(name));
```

Comment:
Same comment as above about `emplace_back()`.
... Ah, but that's because this commit moves that code.

Reply:
OK, please resolve the conversation if you're satisfied.